pkg/kubefed2/util/util.go

```diff
@@ -91,8 +91,12 @@ func HostClientset(config *rest.Config) (*client.Clientset, error) {
 
 // ClusterClientset provides a kubernetes API compliant clientset to
 // communicate with the joining cluster's kubernetes API server.
-func ClusterClientset(config *rest.Config) (*client.Clientset, error) {
-	return client.NewForConfigOrDie(config), nil
+func ClusterClientset(config *rest.Config, returnErr bool) (*client.Clientset, error) {
```

Comment:
I think changing how this client helper works should be tackled separately rather than differentiating just `ClusterClientset` to ensure consistency.